backend/src/apiserver/filter/filter.go

```diff
@@ -124,6 +135,14 @@ func checkPredicate(p *api.Predicate) error {
 			return fmt.Errorf("cannot use scalar operator %v on array type %T", p.Op, t)
 		}
 
+	case api.Predicate_IS_SUBSTRING:
+		switch t := p.Value.(type) {
+		case *api.Predicate_StringValue:
+			return nil
+		default:
+			return fmt.Errorf("cannot use non string value type %T with operator %v", p.Op, t)
```

Comment:
i don't think we should return fmt.errorf, as we dissused before.

Reply:
same applies to other places

Reply:
Done.